crates/node/src/sync/state.rs

```diff
@@ -71,6 +73,17 @@ impl Node {
 
         let mut sqx_out = Sequencer::default();
 
+        let possible_sending_key = self
+            .ctx_manager
+            .get_sender_key(&context.id, &our_identity)?;
+
+        let sending_key = match possible_sending_key {
+            Some(key) => key,
+            None => todo!(),
+        };
```

Comment:
this should match what you did in `blobs`, use the private key, not the sender key